[kohkakohla] iP

kohkakohla opened the pull request:
# BotimusPrime: Your Task Manager 🤖

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BotimusPrime frees your mind of having to remember things you need to do. It's:
- **text-based**
- *easy to learn* (simple commands like `bye`)
- ~~FAST~~ **SUPER FAST** to use
- All you need to do is,

1. **[Download it from here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html)**.
2. *Double-click it.*
3. **Add your tasks**.
4. Let it manage your tasks for you 😉

And it is **FREE**!

## Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Reminders

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        new BotimusPrime("todolist.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kohkakohla committed:
Changed chatbot name, implemented initial skeleton

kohkakohla committed:
BotimusPrime.java: Bot now echoes commands entered by user

kohkakohla committed:
BotimusPrime.java: added ability to store and list items

kohkakohla committed:
Task.java: Added new Task class to encapsulate a task in todolist
BotimusPrime.java: Added ability to mark tasks as done

kohkakohla committed:
Added Deadline, Event, Todo tasks

kohkakohla committed:
Error handling added

kohkakohla committed:
Added deletion of tasks

kohkakohla committed:
Updated Text UI testing

kohkakohla committed:
BotimusPrime class: add save, load
Task classes: add save formatting

kohkakohla committed:
Revert "BotimusPrime class: add save, load"

This reverts commit 6bdfcfffc36ca6cae3bf6f0bc0a19611fdd56cad.

kohkakohla committed:
Deadline.java: Format code

kohkakohla committed:
Merge branch 'branch-Level-7'

kohkakohla committed:
BotimusPrime class: Add save, load

kohkakohla committed:
Merge branch 'branch-Level-7'

kohkakohla committed:
Revert "Merge branch 'branch-Level-7'"

This reverts commit ec4cd5ab6977704ef9e66f9ef02a9f109a79603b, reversing
changes made to d7b6e0b8923bcae8eb8a6f9d2278eb796d745a15.

kohkakohla committed:
Dummy commit to force merge

kohkakohla committed:
Merge branch 'branch-Level-7'

kohkakohla committed:
Merge branch 'branch-Level-7'

kohkakohla committed:
Task classes: Add saving times as DateTime objects

kohkakohla committed:
Merge branch 'branch-Level-8'

kohkakohla committed:
Extract Ui, Storage, Parser, TaskList classes

kohkakohla committed:
Revert "Extract Ui, Storage, Parser, TaskList classes"

This reverts commit e7d6942d73ed4ff50c0369c67a576c6330805ee7.

kohkakohla committed:
Dummy commit to allow merge

kohkakohla committed:
Merge branch 'branch-A-MoreOOP'

kohkakohla committed:
Merge branch 'branch-A-MoreOOP'

kohkakohla committed:
Organise classes into packages

kohkakohla committed:
Merge branch 'branch-A-Packages'

kohkakohla committed:
Merge branch 'add-gradle-support'

kohkakohla committed:
Refactor changes to class names

kohkakohla committed:
Merge branch 'branch-A-Gradle'

kohkakohla committed:
Add tests for TaskList, Storage classes

kohkakohla committed:
Merge branch 'branch-A-JUnit'

kohkakohla committed:
Add JavaDocs to major classes

kohkakohla committed:
Tweak code to comply with coding standards

kohkakohla committed:
TaskList.java: Add find function

kohkakohla committed:
Merge branch 'branch-A-JavaDoc'

kohkakohla committed:
Merge with branch 'branch-A-CodingStandard'

kohkakohla committed:
Merge branch 'branch-Level-9'

LiHaixin000 commented on src/main/java/botimusprime/BotimusPrime.java:
```diff
@@ -0,0 +1,74 @@
+ */
+public class BotimusPrime {
+
+    private Storage storage;
+    private Ui ui;
+    private TaskList taskList;
+    private Parser parser;
```
These field names are appropriately named as they are nouns representing objects or entities. They are concise and descriptive, which is good

LiHaixin000 commented on src/main/java/botimusprime/BotimusPrime.java:
```diff
@@ -0,0 +1,74 @@
+     * Displays a greeting message, then enters a loop to read and parse user commands.
+     * The loop continues until the user issues an exit command "bye".
+     */
+    public void run() {
```
The method name run is a bit generic. While it is commonly used to start processes or applications, a more descriptive name like startChatbot could provide clearer insight into its purpose.

kohkakohla replied:
Sounds good!

LiHaixin000 commented on src/main/java/botimusprime/parser/Parser.java:
```diff
@@ -0,0 +1,100 @@
+            taskList.delete(input);
+        } else if (input.startsWith("find")) {
+            taskList.findTask(input);
+        } else {
+            System.out.println("bro out here speaking nonsense issit");
+        }
```
This error message contains slang and informal language, which is not recommended for professional or production-level code. A more formal message like "Unrecognized command. Please try again." would be more appropriate and user-friendly

LiHaixin000 commented on src/main/java/botimusprime/tasks/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+        return false;
+    }
+
+    @Override
+    public boolean isToDo() {
+        return true;
+    }
+
+    @Override
+    public boolean isDeadline() {
+        return false;
+    }
```
The method isToDo returns true, while isDeadline returns false, which seems contradictory given the class is named Deadline. This might be a mistake.

kohkakohla replied:
Oh my goodness, good spot! I missed this completely, thanks so much

LiHaixin000 reviewed (commented):
LGTM. Just some nits to fix. Good job!

kohkakohla committed:
Add checkstyle

ChinSekYi commented on src/main/java/botimusprime/parser/Parser.java:
```diff
@@ -0,0 +1,101 @@
+        DateTimeFormatter withTime = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        DateTimeFormatter withoutTime = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        DateTimeFormatter slashWithTime = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        DateTimeFormatter slashWithoutTime = DateTimeFormatter.ofPattern("dd/MM/yyyy");
```
I like you have a datetimeformatter for both with time and without time.

ChinSekYi commented on src/main/java/botimusprime/tasks/Event.java:
```diff
@@ -0,0 +1,42 @@
+public class Event extends Task {
+    protected LocalDateTime from;
+    protected LocalDateTime to;
+    public Event(String description, boolean isDone, LocalDateTime from, LocalDateTime to) {
```
You may add a blank line here to separate the attributes and the constructor.

ChinSekYi commented on src/main/java/botimusprime/tasks/Event.java:
```diff
@@ -0,0 +1,42 @@
+    public boolean isDeadline() {
+        return false;
+    }
+
```
You may add a Override tag here. Can do the same for other tasks that extends Task class too.

ChinSekYi reviewed (commented):
Your codes are of good quality, I did not find much code quality issues. I have only raise up a few suggestions upon reviewing your code.

Judy1x4 commented on src/main/java/botimusprime/tasks/TaskList.java:
```diff
@@ -0,0 +1,277 @@
+        tasks.add(task);
+        storage.saveToDisk(this);
+
+        System.out.println(
+                "____________________________________________________________\n" +
+                        String.format("Alright, I've added the task:\n %s\nNow you have %d tasks in the list.\n",
+                                task, tasks.size())
+                                        + "____________________________________________________________\n");
+    }
```
You could add method that returns Strinf for the new line to reduce the code length.

Judy1x4 reviewed (commented):
LGTM. Just nits to fix.

kohkakohla committed:
Add JavaFX GUI

kohkakohla committed:
Merge branch 'branch-Level-10'

kohkakohla committed:
Add assertions to TaskList methods\n\nTaskList methods do not check for valid input, and empty tasklist.\n\nWe now add assertions to make sure input is valid, and tasklist is not\nempty to reduce error.\n\nLet's add assertions to taskList methods.

kohkakohla committed:
Refactor code to improve code quality

The current code handled regex weirdly, and constants were declared all
over the place.

Currently, the code in Parser.java has been refactored to handle user
input more cleanly, with regex patterns.

This helps with overall code understandability, and it is easier to handle
edge cases.

kohkakohla committed:
Merge pull request #1 from kohkakohla/branch-A-Assertions

Add assertions to TaskList methods

kohkakohla committed:
Merge branch 'master' into branch 'branch-A-CodeQuality'

kohkakohla committed:
Merge pull request #2 from kohkakohla/branch-A-CodeQuality

Refactor code to improve code quality

kohkakohla committed:
Add view schedule functionality

kohkakohla committed:
Merge branch 'branch-B-ViewSchedules'

kohkakohla committed:
Edit readme.md in docs

kohkakohla committed:
Add exception handling to Parser and TaskList

kohkakohla committed:
Update README.md

kohkakohla committed:
Update README.md

kohkakohla committed:
Update README.md
